### PR TITLE
Defer running of adBlockInUse

### PR DESCRIPTION
**Pull request by holder-josh**
Fix for a longstanding bug: https://github.com/guardian/frontend/issues/9927

On slower browsers or when opening multiple tabs in Firefox, adBlockInUse would try to modify the dom before the dom was ready. This defers the running of adBlockInUse and stores its value.

@rich-nguyen 

This affects commercial components & tracking, can you take a look @Calanthe @uplne?

### CODE REVIEW

**Comment by mattandrews** [comment]
Ahhh brilliant! Thanks @Jholder112233 and team!

**Comment by davidrapson** [comment]
:gem: :fireworks:

File: static/src/javascripts/projects/common/utils/detect.js [hunk 1]
```diff
@@ -386,10 +386,15 @@ define([
     }
 
     function adblockInUse() {
-        var sacrificialAd = createSacrificialAd();
-        var contentBlocked = isHidden(sacrificialAd);
-        sacrificialAd.remove();
-        return contentBlocked;
+        if (!detect.cachedAdblockInUse) {
+            var sacrificialAd = createSacrificialAd();
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/utils/detect.js]: I thought that we have a convention to use single var?

**Comment by Calanthe** [comment]
Code looks fine. I hope that it still works too ;)

File: static/src/javascripts/projects/common/modules/analytics/omniture.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ define([
         this.s.prop31    = id.getUserFromCookie() ? 'registered user' : 'guest user';
         this.s.eVar31    = id.getUserFromCookie() ? 'registered user' : 'guest user';
 
-        this.s.prop40    = detect.adblockInUse || detect.getFirefoxAdblockPlusInstalled();
+        this.s.prop40    = detect.adblockInUse() || detect.getFirefoxAdblockPlusInstalled();
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/analytics/omniture.js]: nice find

**Comment by rich-nguyen** [comment]
:+1:

**Comment by paperboyo** [comment]
Whoa! :gem:!

**Pull request merged**